src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.cs

```diff
@@ -2168,12 +2168,36 @@ internal void GetNFIValues(NumberFormatInfo nfi)
                 nfi._currencyNegativePattern = GetLocaleInfoCoreUserOverride(LocaleNumberData.NegativeMonetaryNumberFormat);
                 nfi._numberNegativePattern = GetLocaleInfoCoreUserOverride(LocaleNumberData.NegativeNumberFormat);
 
+                nfi._nativeDigits = NumberFormatInfo.s_asciiDigits;
+
                 // LOCALE_SNATIVEDIGITS (array of 10 single character strings).
                 string digits = GetLocaleInfoCoreUserOverride(LocaleStringData.Digits);
-                nfi._nativeDigits = new string[10];
-                for (int i = 0; i < nfi._nativeDigits.Length; i++)
+
+                // if digits.Length < NumberFormatInfo.s_asciiDigits.Length means the native digits setting is messed up in the host machine.
```

Comment:
The current method is kind of doing a lot of work already. What do you think about extracting this logic into a separate method, and then here we would just say:

```C#
nfi._nativeDigits = GetNativeDigits();
```

Reply:
Ok, will try that.